WordPress/src/main/java/org/wordpress/android/ui/comments/unified/UnifiedCommentsEditViewModel.kt

```diff
@@ -0,0 +1,286 @@
+            return
+        }
+
+        _uiState.value?.let { uiState ->
+            val editedContent = uiState.editedComment
+
+            launch(bgDispatcher) {
+                setLoadingState(SAVING)
+
+                val comment = commentsStore.getCommentByLocalId(editedContent.commentId).firstOrNull()
+
+                comment?.let {
+                    val updatedComment = comment.copy(
+                            authorUrl = editedContent.userWebAddress,
+                            authorName = editedContent.userName,
+                            authorEmail = editedContent.userEmail,
+                            content = editedContent.commentText
+                    )
+                    val result = commentsStore.updateEditComment(site, updatedComment)
+
+                    if (result.isError) {
+                        setLoadingState(NOT_VISIBLE)
+                        _onSnackbarMessage.postValue(
+                                Event(SnackbarMessageHolder(UiStringRes(R.string.error_edit_comment)))
+                        )
+                    } else {
+                        _uiActionEvent.postValue(Event(DONE))
+                    }
+                }
+            }
+        }
+    }
+
+    fun onBackPressed() {
+        _uiActionEvent.value = Event(CLOSE)
+    }
+
+    private fun initViews(commentId: Int) {
+        launch {
+            setLoadingState(LOADING)
+
+            val commentList = withContext(bgDispatcher) {
+                commentsStore.getCommentByLocalId(commentId.toLong())
+            }
+
+            if (commentList.isEmpty()) {
+                _onSnackbarMessage.value = Event(SnackbarMessageHolder(
+                        message = UiStringRes(R.string.error_load_comment),
+                        onDismissAction = { _ ->
+                            _uiActionEvent.value = Event(CLOSE)
+                        }
+                ))
+                return@launch
+            } else {
+                val comment = commentList.first()
+                val commentEssentials = CommentEssentials(
+                        commentId = comment.id,
+                        userName = comment.authorName ?: "",
+                        commentText = comment.content ?: "",
+                        userWebAddress = comment.authorUrl ?: "",
+                        userEmail = comment.authorEmail ?: ""
+                )
+
+                _uiState.value = EditCommentUiState(
+                        isMenuEnabled = false,
+                        shouldInitComment = true,
+                        shouldInitWatchers = true,
+                        showProgress = LOADING.show,
+                        progressText = LOADING.progressText,
+                        originalComment = commentEssentials,
+                        editedComment = commentEssentials,
+                        editErrorStrings = EditErrorStrings()
+                )
+            }
+
+            delay(LOADING_DELAY_MS)
```

Comment:
This is again more of a question than a change request: Why do we need the delay here since the process seems to be really fast? Is it just for UI/animation purposes?

Reply:
Yep, that is more to give a sequencing to the logic on the UI, under some circumstances it happened to me during testing that if the process is quick but not quick enough it seems like you can see a flashing effect on the UI where the progress screen appear for a quick instant and it can seem more like a glitch; so I thought this was looking generally better (hopefully :) )